### PR TITLE
SW4: new package

### PR DESCRIPTION
**Pull request by houjun**
[SW4](https://github.com/geodynamics/sw4) is a program for simulating seismic wave propagation on parallel computers.

### CODE REVIEW

**Review by andersp — DISMISSED** [review]

File: var/spack/repos/builtin/packages/sw4/package.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+
+class Sw4(MakefilePackage):
+    """This package builds SW4 with MPI, OpenMP, HDF5, FFTW, PROJ, and ZFP."""
+
+    homepage = "https://github.com/geodynamics/sw4"
+    git = "https://github.com/geodynamics/sw4.git"
+
+    maintainers = ["houjun", "andersp"]
+    # tags = ["e4s"]
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/sw4/package.py]: Do you want to use a different tag or list of tags here?

File: var/spack/repos/builtin/packages/sw4/package.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    version("3.0-beta2", tag="v3.0-beta2")
+    version("master", branch="master")
+    version("developer", branch="developer")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/sw4/package.py:20]: FWIW. The "convention" is branch versions are listed before numbered versions.

**Review by tldahlgren — COMMENTED** [review]
Confirmed the three versions exist in the repo.  Just a few minor questions.

File: var/spack/repos/builtin/packages/sw4/package.py (hunk 1 above)
Comment (houjun) [reply var/spack/repos/builtin/packages/sw4/package.py]: I removed the unnecessary tags line, will add it in later versions

File: var/spack/repos/builtin/packages/sw4/package.py (hunk 2 above)
Comment (houjun) [reply var/spack/repos/builtin/packages/sw4/package.py:20]: fixed

**Review by tldahlgren — APPROVED** [review]
LGTM

**Review by andersp — APPROVED** [review]

**Pull request merged**